### ISSUE TITLE
Can't get stateful redis cluster connection, when the first node of redis cluster is unavailable.

### ISSUE THREAD

**Issue by leofanaq** [mp911de/lettuce#164]
Hi mp911de, My redis cluster looks like this:
```
Partitions [
RedisClusterNode [uri=RedisURI [host='192.168.0.122', port=6380], nodeId='071cbfb1d0706c7f5db1250f52d5fdd2c8184c56', connected=true, slaveOf='null', pingSentTimestamp=0, pongReceivedTimestamp=0, configEpoch=1, flags=[MYSELF, MASTER], slot count=5461], 
RedisClusterNode [uri=RedisURI [host='192.168.0.123', port=6380], nodeId='2f17db2a1b066bbef67339e12500e4ea3fdbd0cd', connected=true, slaveOf='9dc33d837197d9fa919e404f8580afa184821c4a', pingSentTimestamp=0, pongReceivedTimestamp=1450618611017, configEpoch=8, flags=[SLAVE], slot count=0], 
RedisClusterNode [uri=RedisURI [host='192.168.0.123', port=6381], nodeId='9349de7d551a5e0ae2b73e5b0edf7efca2d28f26', connected=true, slaveOf='071cbfb1d0706c7f5db1250f52d5fdd2c8184c56', pingSentTimestamp=0, pongReceivedTimestamp=1450618610008, configEpoch=5, flags=[SLAVE], slot count=0], 
RedisClusterNode [uri=RedisURI [host='192.168.0.123', port=6382], nodeId='f2cef699566014933f69ee3d61fba45ce3493d89', connected=true, slaveOf='2d8c8a96aa41f62399f10940f079ef99b1f58bdf', pingSentTimestamp=0, pongReceivedTimestamp=1450618612026, configEpoch=6, flags=[SLAVE], slot count=0], 
RedisClusterNode [uri=RedisURI [host='192.168.0.122', port=6382], nodeId='9dc33d837197d9fa919e404f8580afa184821c4a', connected=true, slaveOf='null', pingSentTimestamp=0, pongReceivedTimestamp=1450618609004, configEpoch=8, flags=[MASTER], slot count=5462], 
RedisClusterNode [uri=RedisURI [host='192.168.0.122', port=6381], nodeId='2d8c8a96aa41f62399f10940f079ef99b1f58bdf', connected=true, slaveOf='null', pingSentTimestamp=0, pongReceivedTimestamp=1450618613033, configEpoch=2, flags=[MASTER], slot count=5461]]
```

Redis client like this:
```java
        // Syntax: redis://[password@]host[:port]
    	List<RedisURI> redisURIs = new ArrayList<RedisURI>();
    	redisURIs.add(RedisURI.create("redis://192.168.0.122:6380"));
    	redisURIs.add(RedisURI.create("redis://192.168.0.122:6381"));
    	redisURIs.add(RedisURI.create("redis://192.168.0.122:6382"));
    	redisURIs.add(RedisURI.create("redis://192.168.0.123:6380"));
    	redisURIs.add(RedisURI.create("redis://192.168.0.123:6381"));
    	redisURIs.add(RedisURI.create("redis://192.168.0.123:6382"));
        RedisClusterClient redisClient = RedisClusterClient.create(redisURIs);
        redisClient.setOptions(new ClusterClientOptions.Builder()
                .refreshClusterView(true)
                .refreshPeriod(1, TimeUnit.MINUTES)
                .build());
        System.out.println(redisClient.getPartitions().toString());
        StatefulRedisClusterConnection<String, String> connection = redisClient.connect();

        System.out.println("Connected to Redis");

        connection.close();
        redisClient.shutdown();
```

When 192.168.0.122:6380 is running, this client works well. 
When I killed it, this client get error message:
```
Partitions [RedisClusterNode [uri=RedisURI [host='192.168.0.123', port=6380], nodeId='2f17db2a1b066bbef67339e12500e4ea3fdbd0cd', connected=true, slaveOf='9dc33d837197d9fa919e404f8580afa184821c4a', pingSentTimestamp=0, pongReceivedTimestamp=1450627345833, configEpoch=8, flags=[SLAVE], slot count=0], RedisClusterNode [uri=RedisURI [host='192.168.0.123', port=6381], nodeId='9349de7d551a5e0ae2b73e5b0edf7efca2d28f26', connected=true, slaveOf='null', pingSentTimestamp=0, pongReceivedTimestamp=1450627349868, configEpoch=9, flags=[MASTER], slot count=5461], RedisClusterNode [uri=RedisURI [host='192.168.0.122', port=6382], nodeId='9dc33d837197d9fa919e404f8580afa184821c4a', connected=true, slaveOf='null', pingSentTimestamp=0, pongReceivedTimestamp=1450627348860, configEpoch=8, flags=[MASTER], slot count=5462], RedisClusterNode [uri=RedisURI [host='192.168.0.123', port=6382], nodeId='f2cef699566014933f69ee3d61fba45ce3493d89', connected=true, slaveOf='null', pingSentTimestamp=0, pongReceivedTimestamp=1450627346843, configEpoch=10, flags=[MASTER], slot count=5461], RedisClusterNode [uri=RedisURI [host='192.168.0.122', port=6381], nodeId='2d8c8a96aa41f62399f10940f079ef99b1f58bdf', connected=true, slaveOf='f2cef699566014933f69ee3d61fba45ce3493d89', pingSentTimestamp=0, pongReceivedTimestamp=0, configEpoch=2, flags=[SLAVE, MYSELF], slot count=0], RedisClusterNode [uri=RedisURI [host='192.168.0.122', port=6380], nodeId='071cbfb1d0706c7f5db1250f52d5fdd2c8184c56', connected=false, slaveOf='9349de7d551a5e0ae2b73e5b0edf7efca2d28f26', pingSentTimestamp=1450627342494, pongReceivedTimestamp=1450627339768, configEpoch=9, flags=[SLAVE], slot count=0]]
Exception in thread "main" com.lambdaworks.redis.RedisConnectionException: Unable to connect to /192.168.0.122:6380
	at com.lambdaworks.redis.AbstractRedisClient.initializeChannel(AbstractRedisClient.java:216)
	at com.lambdaworks.redis.AbstractRedisClient.connectAsyncImpl(AbstractRedisClient.java:101)
	at com.lambdaworks.redis.cluster.RedisClusterClient.connectClusterImpl(RedisClusterClient.java:333)
	at com.lambdaworks.redis.cluster.RedisClusterClient.connect(RedisClusterClient.java:194)
	at com.lambdaworks.redis.cluster.RedisClusterClient.connect(RedisClusterClient.java:181)
	at com.beautyorange.test.redis.lettuceClient.App.main(App.java:37)
Caused by: java.net.ConnectException: Connection refused: /192.168.0.122:6380
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:224)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:289)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:528)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:110)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137)
	at java.lang.Thread.run(Thread.java:745)
```

At the same time, run redis-trib.rb check 192.168.0.122:6381, other nodes work well and all slots covered. How to fix it?

**Comment by mp911de** [comment]
@leofanaq lettuce fails when the first node in the seed list or the partition list (once it is loaded) is down. Currently there is no workaround. The feature to skip unhealthy nodes at connect time is overdue and I want to implement that feature.

**Comment by mp911de** [comment]
This issue is fixed now and the fix is available as snapshot (4.1-SNAPSHOT) on Sonatype's Snapshot repo (https://oss.sonatype.org/content/repositories/snapshots/).

**Comment by leofanaq** [comment]
thanks
